rustls/src/conn.rs

```diff
@@ -946,7 +947,11 @@ impl ConnectionCommon {
     /// buffering, so `rd` can supply TLS messages in arbitrary-
     /// sized chunks (like a socket or pipe might).
     pub(crate) fn read_tls(&mut self, rd: &mut dyn io::Read) -> io::Result<usize> {
-        self.message_deframer.read(rd)
+        let res = self.message_deframer.read(rd);
```

Comment:
Probably the `read()` should not happen here if `self.peer_eof` is already `true`? The peer may not immediately terminate the TCP connection after sending a `CloseNotify`, and `read()` here will return `WantRead` error, leading to the caller waiting in `Pending` instead of getting a `ConnectionAborted` error with `CloseNotify alert received` as the message.